src/Nethermind/Nethermind.TxPool/ITxPool.cs

```diff
@@ -15,11 +15,17 @@ public interface ITxPool
         Transaction[] GetPendingTransactions();
 
         /// <summary>
-        /// Grouped by sender address, sorted by nonce and later tx pool sorting
+        /// Non-blob txs grouped by sender address, sorted by nonce and later tx pool sorting
         /// </summary>
         /// <returns></returns>
         IDictionary<Address, Transaction[]> GetPendingTransactionsBySender();
 
+        /// <summary>
+        /// Lazy enumerate blob txs starting from the best one
+        /// </summary>
+        /// <returns></returns>
+        IEnumerable<Transaction> GetPendingBlobTransactions();
```

Comment:
The difference here is that we have normal txs in full form in memory and we can easily expose it to `TxPoolTxSource`, and for blob txs we have in memory only light equivalences without actual blobs, so we need to load them from db before exposing. If we would like to act like with normal txs, we would need to load from db all blob txs, so it can be even 16k db reads. In current design, we are operating on light tx equivalences stored in memory to pick best tx and then loading full tx from db. As we need only max 6 blob txs per block, there are max 6 db reads per block preparation attempt

Reply:
It's refactored as requested, just working on collection of light blob txs instead of full ones